File: test/howToF#.Tests/CommandTests.fs
module Test

open System
open Xunit
open FsUnit.Xunit
open howTo.Models


type CommandTests() =
    let mutable testCommand = null
    do testCommand <- Commands(HowTo = "Do something awesome", Platform = "xUnit", CommandLine = "dotnet test")

    [<Fact>]
    member __.``Can Change HowTo``() =
        //Act
        testCommand.HowTo <- "Execute Unit Tests"
        // Assert
        testCommand.HowTo
        |> should equal "Execute Unit Tests"

    [<Fact>]
    member __.``Can change platform``() =
        testCommand.Platform <- "xUnit"

        testCommand.Platform |> should equal "xUnit"

    interface IDisposable with
        member this.Dispose() = testCommand <- null

File: test/howTo.Tests/CommandTests.cs
using System;
using Xunit;
using howTo.Models;

namespace howTo.Tests
{
    public class CommandTests : IDisposable
    {
        Commands? testCommand;
        public CommandTests()
        {
            testCommand = new Commands
            {
                HowTo = "Do something awesome",
                Platform = "xUnit",
                CommandLine = "dotnet test"
            };
        }
        public void Dispose()
        {
            testCommand = null;
        }
        [Fact]
        public void CanChangeHowTo()
        {
            //Arrange
            //Act
            testCommand!.HowTo = "Execute Unit Tests";

            //Assert
            Assert.Equal("Execute Unit Tests", testCommand.HowTo);
        }
    }
}
